### DISCUSSION TITLE
PSX/PS2 PCB Issue

### DISCUSSION THREAD

**Discussion by pmgducati** [darthcloud/BlueRetro#234] [Q&A] [answered]
Hey @darthcloud,
I have been working through all the new PCBs I have been working on and everything (N64,NES, Gamecube, 3DO and PCE) all work perfectly with the exception the the PSX/PS2.  It powers on and I can connect the controllers to the ESP32 but no buttons work.  I'm using FW v1.4.

Here is the Schematic, Layout and a couple images of the PCB:
![PS2 2](https://user-images.githubusercontent.com/20894227/159360289-b8572fee-1bcd-4f9d-9fc8-8f444f6e9b5e.png)
![PS2 1](https://user-images.githubusercontent.com/20894227/159360300-38b7c5c3-de00-4304-a63e-b81b6367f309.png)
![PXL_20220321_204347802](https://user-images.githubusercontent.com/20894227/159361047-08cd04cc-8f27-4fa9-9148-2f9324f260c1.jpg)
![PXL_20220321_204408090](https://user-images.githubusercontent.com/20894227/159361062-fa602d94-7ea8-42bf-a1b2-3a2d4648fb64.jpg)
![PXL_20220321_204356610](https://user-images.githubusercontent.com/20894227/159361058-8f160dff-b17c-400a-af5d-b57a8c53cc38.jpg)

Any Thoughts?

**Comment by darthcloud** [comment]
your DB25 pin number are mirrored, normal? db25 pin one is IO23

**Reply by darthcloud** [reply]
I guess normal since you used female symbol instead of male.

**Reply by pmgducati** [reply]
Yeah I put the female and on all of the system specific sides so I only have to deal with one set of pins if they get bent.

**Comment by darthcloud** [comment]
Hard to tell, try hardcoded firmware

**Reply by pmgducati** [reply]
I'll give that a try tonight

**Reply by pmgducati** [reply]
I used the PSX/PS2 firmware (1.4) and everything works fine.  There seems to be an issue with the universal FW identifying as a PSX/PS2.  Am I correct you do this with the pin combination tied to GND or 3V3?

**Reply by darthcloud** [reply]
This was just me having a brain fart back in 2019, PSX and PS2 are same thing, so I39 down or low is the same thing.

The important part is the detections pins: IO19, IO21, IO22, IO25, IO32, IO33, I34 and I35.

The first of those pins that move will determinate which consoles is used. For PSX/2 it's I34. 

Disconnect the 8V supply from console and use usb to power and get the log while using universal FW:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

The log will tell which system it detect and help figure what is wrong in wiring.

**Reply by pmgducati** [reply]
I have attached the log, it looks like it thinks it is a PC-FX

[psx.txt](https://github.com/darthcloud/BlueRetro/files/8326797/psx.txt)

**Reply by darthcloud** [reply]
This means it see IO33 moving before I34 does, for PS I34 is the chip select (aka DTR or ATT), and IO33 is the clock.

The DTR line should always goes low before the clk signal start. Double check those connection, maybe you got a resistive short that glitch IO33 at boot.

**Reply by pmgducati** [reply]
After some experimenting I went through previous discussions and found this:

https://github.com/darthcloud/BlueRetro/discussions/87

He was having the same PC-FX issue that I was having towards the end of the discussion, I tried the solve of disconnection the 8V from the controller and connecting the 3V3 to 3V3 on the ESP instead.  Now everything works perfectly!  I have tried about a dozen games, rebooting the PS2 and the ESP with no issues.  Would it be worth suggesting in your schematic that if using a ESP32 devkit, omit the 8V pin and use 3v3 instead?

**Reply by darthcloud** [reply]
For some the 3V3 is too weak, and for some the 8V is bad somehow. I need to investigate a solution.

**Comment by pmgducati** [comment, marked as answer]
@darthcloud Thanks for the help, I added the 8V Pad along with the 3V3 pad and a note to only use one, so there is the option to figure out what will work for the best for each console.

Thanks for you help!
